### DISCUSSION TITLE
Help switch for all playlist content

### DISCUSSION THREAD

**Discussion by wezo3** [savonet/liquidsoap#2494] [General]
Hello,
when i use switch feature on the time for playlist I see its play for maximum 1 min and after complete the song its back again to the other playlist in the fallback 

for example :

```
normalp = request.dynamic(id="radio",next_song)
jingle = playlist(mode="normal", reload_mode="watch", "/home/playlist.txt")
jingle = switch([({ 15m }, jingle)])
input = fallback(track_sensitive = false, [jingle, normalp ])
```

the text file /home/playlist.txt contain more than 6 mp3 files for total duration 3.34 min
if the duration for first song less than 1 min its complete and start for the second ..etc until 1 minute has passed after that not resume the jingle playlist, it fallback again to normalp playlist

how can i play the jingle playlist at the time with all content until finish and switch to normalp playlist after complete ?

Regards

**Comment by toots** [comment]
Hi,

The predicate for your switch is only valid on the 15th minute of every hour, so the script is behaving like expected. I would suggest to take a look at the Liquidsoap book here: https://www.liquidsoap.info/doc-dev/book.html it should have more details and explanations about how these operators work and how to use them to suit your needs.

**Comment by wezo3** [comment]
Thank you for your response 
I already checked the book its great but this point still not find solution for it
I tried to use this logic but there something if the playlist total duration more than this value will cut without resume until complete 
loop=false and 15m-25m

jingle = playlist(loop=false, mode="normal", reload_mode="watch", "/home/playlist.txt")
jingle = switch([({ 15m-25m }, jingle)])

its works good if the total playlist tracks duration less than 10 min but if more than 10 min will cut 
just i need the all playlist tracks content play every hour and start on min 15 until playlist complete 

Thanks
